src/components/refactor/dialogs/generator-creation/reactive-capability-curve/reactive-capability-utils.js

```diff
@@ -0,0 +1,88 @@
+import {
+    toNumber,
+    validateValueIsANumber,
+} from '../../../../util/validation-functions';
+
+const reactiveCapabilityCurveRowEmptyFormData = {
+    [P]: null,
+    [Q_MAX_P]: null,
+    [Q_MIN_P]: null,
+};
+export const getReactiveCapabilityCurveEmptyFormData = (
+    id = REACTIVE_CAPABILITY_CURVE_TABLE
+) => ({
+    [id]: [
+        reactiveCapabilityCurveRowEmptyFormData,
+        reactiveCapabilityCurveRowEmptyFormData,
+    ],
+});
+
+export const getReactiveCapabilityCurveValidationSchema = (
+    id = REACTIVE_CAPABILITY_CURVE_TABLE
+) => ({
+    [id]: yup
+        .array()
+        .nullable()
+        .when([REACTIVE_CAPABILITY_CURVE_CHOICE], {
+            is: 'CURVE',
+            then: (schema) =>
+                schema
+                    .of(
+                        yup.object().shape({
+                            [Q_MAX_P]: yup.number().nullable().required(),
+                            [Q_MIN_P]: yup
+                                .number()
+                                .nullable()
+                                .required()
+                                .lessThan(
+                                    yup.ref(Q_MAX_P),
+                                    'ReactiveCapabilityCurveCreationErrorQminPQmaxPIncoherence'
+                                ),
+                            [P]: yup
+                                .number()
+                                .nullable()
+                                .required()
+                                .min(
+                                    yup.ref(Q_MIN_P),
+                                    'ReactiveCapabilityCurveCreationErrorPOutOfRange'
+                                )
+                                .max(
+                                    yup.ref(Q_MAX_P),
+                                    'ReactiveCapabilityCurveCreationErrorPOutOfRange'
+                                ),
+                        })
+                    )
+                    .min(2, 'ReactiveCapabilityCurveCreationErrorMissingPoints')
+                    .test(
+                        'validateP',
+                        'ReactiveCapabilityCurveCreationErrorPInvalid',
+                        (values) => {
+                            const everyValidP = values
+                                .map((element) =>
+                                    // Note : convertion toNumber is necessary here to prevent corner cases like if
+                                    // two values are "-0" and "0", which would be considered different by the Set below.
+                                    validateValueIsANumber(element.p)
+                                        ? toNumber(element.p)
+                                        : null
+                                )
+                                .filter((p) => p !== null);
+                            const setOfPs = [...new Set(everyValidP)];
+                            return setOfPs.length === everyValidP.length;
```

Comment:
make a function to explain what we are checking here.
From what I understand, we are checking if the not null values among the array are all distinct